aiida/common/archive.py

```diff
@@ -284,39 +283,46 @@ def extract_tar(infile, folder, nodes_export_subfolder="nodes", silent=False):
     :param folder: a SandboxFolder, used to extract the file tree
     :param nodes_export_subfolder: name of the subfolder for AiiDA nodes
     :param silent: suppress debug print
+    :raises `CorruptArchive`: if the archive misses files or files have incorrect formats
     """
     # pylint: disable=fixme
     if not silent:
-        print("READING DATA AND METADATA...")
+        print('READING DATA AND METADATA...')
 
     try:
-        with tarfile.open(infile, "r:*", format=tarfile.PAX_FORMAT) as handle:
+        with tarfile.open(infile, 'r:*', format=tarfile.PAX_FORMAT) as handle:
+
+            try:
+                handle.extract(path=folder.abspath, member=handle.getmember('metadata.json'))
+            except KeyError:
+                raise CorruptArchive('corrupt archive: required file `metadata.json` is not included')
 
-            handle.extract(path=folder.abspath, member=handle.getmember('metadata.json'))
-            handle.extract(path=folder.abspath, member=handle.getmember('data.json'))
+            try:
+                handle.extract(path=folder.abspath, member=handle.getmember('data.json'))
+            except KeyError:
+                raise CorruptArchive('corrupt archive: required file `data.json` is not included')
 
             if not silent:
-                print("EXTRACTING NODE DATA...")
+                print('EXTRACTING NODE DATA...')
 
             for member in handle.getmembers():
                 if member.isdev():
                     # safety: skip if character device, block device or FIFO
-                    print("WARNING, device found inside the import file: {}".format(member.name), file=sys.stderr)
+                    print('WARNING, device found inside the import file: {}'.format(member.name), file=sys.stderr)
                     continue
                 if member.issym() or member.islnk():
                     # safety: in export, I set dereference=True therefore
                     # there should be no symbolic or hard links.
-                    print("WARNING, link found inside the import file: {}".format(member.name), file=sys.stderr)
+                    print('WARNING, link found inside the import file: {}'.format(member.name), file=sys.stderr)
                     continue
-                # Check that we are only exporting nodes within
-                # the subfolder!
+                # Check that we are only exporting nodes within the subfolder!
                 # TODO: better check such that there are no .. in the
                 # path; use probably the folder limit checks
                 if not member.name.startswith(nodes_export_subfolder + os.sep):
                     continue
                 handle.extract(path=folder.abspath, member=member)
     except tarfile.ReadError:
-        raise ValueError("The input file format for import is not valid (1)")
+        raise ValueError('The input file format for import is not valid (1)')
```

Comment:
do we know what this (1) refers to?

Reply:
I have no idea. These functions I kept as is when I wrote the `Archive` class and so this was already there.